compiler/rustc_hir_typeck/src/gather_locals.rs

```diff
@@ -64,21 +64,46 @@ pub(super) struct GatherLocalsVisitor<'a, 'tcx> {
 }
 
 impl<'a, 'tcx> GatherLocalsVisitor<'a, 'tcx> {
-    pub(super) fn new(fcx: &'a FnCtxt<'a, 'tcx>) -> Self {
-        Self { fcx, outermost_fn_param_pat: None }
+    pub(crate) fn gather_from_local(fcx: &'a FnCtxt<'a, 'tcx>, local: &'tcx hir::LetStmt<'tcx>) {
```

Comment:
pls add a comment to `GatherLocalsVisitor`

- its purpose: it computes the initial info for locals introduced by a given HIR node and adds `Sized` bounds for them
- this visitor must access each local exactly once. Missing any local or visiting it twice will ICE during writeback